advisor/src/main/kotlin/Advisor.kt

```diff
@@ -47,10 +48,10 @@ class Advisor(
         private val LOADER = ServiceLoader.load(AdviceProviderFactory::class.java)
 
         /**
-         * The set of all available [advice provider factories][AdviceProviderFactory] in the classpath, sorted by name.
+         * All [advice provider factories][AdviceProviderFactory] available in the classpath, associated by their names.
          */
-        val ALL: Set<AdviceProviderFactory> by lazy {
-            LOADER.iterator().asSequence().toSortedSet(compareBy { it.providerName })
+        val ALL: SortedMap<String, AdviceProviderFactory> by lazy {
+            LOADER.iterator().asSequence().associateByTo(sortedMapOf(String.CASE_INSENSITIVE_ORDER)) { it.providerName }
```

Comment:
I believe we should fail if duplicate names are used.
E.g. add a require block.

Reply:
That, however, would be new feature; we didn't fail in the corresponding code from `AdvisorCommand` before either. But I agree that we should forbid different plugins of the same type and name on the classpath, and will consider creating another PR to address that.

Reply:
Well, before this change having different provider with the same name resulted in all providers being used. As of this change, providers of duplicate name get silently discarded. That's a behavioral change.

Reply:
> Well, before this change having different provider with the same name resulted in all providers being used.

How? In the former `allVulnerabilityProvidersByName` map, providers were also deduplicated by name.

Reply:
You're right. I've missed that the de-duplicated did exist before.